web/src/inbox_ui.ts

```diff
@@ -766,6 +826,27 @@ function get_sorted_row_dict<T extends DirectMessageContext | TopicContext>(
     return new Map([...row_dict].sort(([, a], [, b]) => b.latest_msg_id - a.latest_msg_id));
 }
 
+function sort_channel_folders(): void {
+    const sorted_channel_folders = [...channel_folders_dict.values()].sort((a, b) => {
+        // Sort OTHER_CHANNELS_FOLDER_ID last, then by name with PINNED_CHANNEL_FOLDER_ID first.
+        if (a.id === OTHER_CHANNELS_FOLDER_ID) {
+            return 1;
+        }
+        if (b.id === OTHER_CHANNELS_FOLDER_ID) {
+            return -1;
+        }
+        if (a.id === PINNED_CHANNEL_FOLDER_ID) {
+            return -1;
+        }
+        if (b.id === PINNED_CHANNEL_FOLDER_ID) {
+            return 1;
+        }
+        return util.strcmp(a.name, b.name);
```

Comment:
I think it'd be best to have this call a new `channel_folders.compare_folders()` function that is this `strcmp` thing, so that this code just works if/when we change the sort key on them. Minor.